Backend: finish /videos route

Edmond120 opened the pull request:
closes #7
closes #10
closes #11

Edmond120 committed:
Server: Minor refactoring

- Improve makefile's clean and test
- Remove useless import in hello_world_test.py
- Update readme.txt to refer to makefile for dependencies

Edmond120 committed:
Server: Add new dependency feedparser and tests

Edmond120 committed:
Server: Add use virtualenv instead of python -m venv

The virtualenv command works for both Mac and Linux.

Edmond120 committed:
Server: remove inaccurate part of a doc string

Edmond120 committed:
Server: refactor routes into seperate file

Edmond120 committed:
Server: add basic watchlist parsing

Edmond120 committed:
Server: allow watchlist configuration option

Edmond120 committed:
Server: Create watchlist if none exists

Edmond120 committed:
Server: start work on rss module

Edmond120 committed:
Server: add omissions in coverage report

Edmond120 committed:
Server: allow rss module to fetch from internet

Edmond120 committed:
Server: new route GET /videos

Edmond120 committed:
Server: change route /videos to /videos_raw

Edmond120 committed:
Server: remove unnecessary print statment

Edmond120 committed:
Server: start adding support for invidious rss feeds

Edmond120 committed:
Server: rename incorrect filetype extension

Edmond120 committed:
Server: add invidious rss feed tests and data

Edmond120 committed:
Server: make rss source types case insensitive

Edmond120 committed:
Server: Use ini files for watchlist + more

Watchlists now use the ini file format. This allows values such as
"source" and "site" to be stored in each entry instead of just the
channel id.

Invidious rss feeds are supported in addition to youtube rss feeds.

Edmond120 committed:
Server: add tests to improve test coverage

coverage is now at 100%

Edmond120 committed:
Server: Refactor watchlist.py just a bit

Edmond120 committed:
Server: mark watchlists with invalid entries

Edmond120 committed:
Server: rename default watchlist to watchlist.ini

Edmond120 committed:
Server: use testing watchlist as default watchlist

Edmond120 committed:
Server: Change default watchlist to only use youtube

Edmond120 committed:
Server: Change default watchlist code

Edmond120 committed:
Server: refactor rss.py and watchlist.py

Edmond120 committed:
Server: remove print statment in watchlist.py

Edmond120 committed:
Server: start work on feed summary

Edmond120 committed:
Server: rename summarize function

Edmond120 committed:
Server: fix spelling error

Edmond120 committed:
Server: organize /videos route output

As of this commit /videos is still unfinished

Edmond120 committed:
Server: complete /videos, incomplete tests

Edmond120 committed:
Server: finish up /videos route

Edmond120 committed:
Server: mark lines to not count for coverage

aziz512 commented on server/app/rss.py:
```diff
@@ -0,0 +1,101 @@
+
+def summarize_entry(entry, raise_error=False):
+	summary = {}
+	def reword(summary_path, entry_path, optional=False):
```
duplicate definition of the same function

aziz512 replied:
I understood by looking into it, but would also be great to get a little comment explaining what it does

aziz512 replied:
@Edmond120  Saw you made a commit in response to this. I was referring to `reword` function specifically, it's defined twice in `summarize_entry` and `summarize_feed`. We can make them into one function in the outer scope if we use more general argument names.

Edmond120 replied:
About that, both reword definitions are closures and since they capture a different state, they are not really duplicates.
I will still make them into one function though since I've noticed that after a change _flatten() is no longer doing any flattening.

aziz512 replied:
Yeah they access (more like populate) the `summary` variable but the function can just as well define it inside and return it. That would actually be better since `reword` would become a pure function.

Edmond120 replied:
@aziz512 done

aziz512 commented on server/app/watchlist.py:
```diff
@@ -0,0 +1,88 @@
+				self.__is_invalid = True
+				continue
+
+			if 'site' not in entry:
```
what's the difference between source and site and why do we need to store both?

Edmond120 replied:
It's mainly there for invidious. Since anyone can host their own instance.

Edmond120 committed:
Server: make thumbnails optional

This is to avoid breaking in the situation where the video thumbnail
can't be found.

Edmond120 committed:
Server: add short comment to rss.py

Edmond120 committed:
Server: refactor _reword() and _flatten() function

Edmond120 committed:
Merge branch 'main' into backend

aziz512 reviewed (approved):
🔥
